clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -250,25 +252,23 @@ public boolean isInErrorState() {
         return currentState == State.ERROR || currentState == State.FENCED;
     }
 
-    public synchronized void setError(Exception exception) {
+    public synchronized void setError(RuntimeException exception) {
         if (exception instanceof ProducerFencedException)
             transitionTo(State.FENCED, exception);
         else
             transitionTo(State.ERROR, exception);
     }
 
-    boolean maybeTerminateRequestWithError(TxnRequestHandler requestHandler) {
-        if (isInErrorState() && requestHandler.isEndTxn()) {
-            // We shouldn't terminate abort requests from error states.
-            EndTxnHandler endTxnHandler = (EndTxnHandler) requestHandler;
-            if (endTxnHandler.builder.result() == TransactionResult.ABORT)
-                return false;
-            String errorMessage = "Cannot commit transaction because at least one previous transactional request " +
-                    "was not completed successfully.";
-            if (lastError != null)
-                requestHandler.fatal(new KafkaException(errorMessage, lastError));
-            else
-                requestHandler.fatal(new KafkaException(errorMessage));
+    private boolean maybeTerminateRequestWithError(TxnRequestHandler requestHandler) {
```

Comment:
nit: In this class, all the private methods come after the public and package-private methods. It would be nice to maintain that consistency.